packages/editor/src/components/collab-sidebar/index.js

```diff
@@ -0,0 +1,303 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect, useDispatch, resolveSelect } from '@wordpress/data';
+import { useState, useEffect, useMemo, useCallback } from '@wordpress/element';
+import { comment as commentIcon } from '@wordpress/icons';
+import { addFilter } from '@wordpress/hooks';
+import { store as noticesStore } from '@wordpress/notices';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import PluginSidebar from '../plugin-sidebar';
+import { collabSidebarName } from './constants';
+import { Comments } from './comments';
+import { AddComment } from './add-comment';
+import { store as editorStore } from '../../store';
+import AddCommentButton from './commentButton';
+import AddCommentToolbarButton from './commentButtonToolbar';
+
+const isBlockCommentExperimentEnabled =
+	window?.__experimentalEnableBlockComment;
+const modifyBlockCommentAttributes = ( settings, name ) => {
+	if ( name?.includes( 'core/' ) && ! settings.attributes.blockCommentId ) {
+		settings.attributes = {
+			...settings.attributes,
+			blockCommentId: {
+				type: 'number',
+				default: 0,
+			},
+		};
+	}
+
+	return settings;
+};
+
+// Apply the filter to all core blocks
+addFilter(
+	'blocks.registerBlockType',
+	'block-comment/modify-core-block-attributes',
+	modifyBlockCommentAttributes
+);
+
+/**
+ * Renders the Collab sidebar.
+ */
+export default function CollabSidebar() {
+	const { createNotice } = useDispatch( noticesStore );
+	const { saveEntityRecord, deleteEntityRecord } = useDispatch( coreStore );
+	const { getEntityRecords } = resolveSelect( coreStore );
```

Comment:
We must use the React hooks so that we get the selectors from the correct provider.